src/ctelnet.cpp

```diff
@@ -642,17 +642,12 @@ bool cTelnet::sendData(QString& data, const bool permitDataSendRequestEvent)
 
     if (mpHost->mAllowToSendCommand) {
         std::string outData;
-        auto errorMsgTemplate = "[ WARN ]  - Invalid characters in outgoing data, one or more characters cannot\n"
-            "be encoded into the range that is acceptable for the character\n"
-            "encoding that is currently set {\"%1\"} for the game server.\n"
-            "It may not understand what is sent to it.\n"
-            "Note: this warning will only be issued once, even if this happens again, until\n"
-            "the encoding is changed.";
+        auto errorMsgTemplate = "[ WARN ]  - Tried to send '%1' to the game, but it is unlikely to understand it.";
```

Comment:
That is fundamentally misleading - how about the slightly longer but much briefer that the original:
```suggestion
        auto errorMsgTemplate = "[ WARN ]  - Tried to send '%1' to the game, but it contains a character that cannot currently be sent over the link between Mudlet and the Game.";
```
Whilst I don't want to drag this out - how can we convey the last sentence, i.e with something about "This is the only warning you will get." - I am thinking perhaps we could light up a :warning: icon in the command line with a tool tip that contain something like the original text...

Reply:
> light up a ⚠️ icon in the command line with a tool tip

we never do it like that, and it's bad for a11y

Reply:
How to flag then that there is a potentially *on-going* **data loss** - if we are shortening the message to not mention that it is the only time the warning will be given...

Reply:
This was the case for a decade before the warning - it's not a practical problem people face to be honest

Reply:
Before the warning we did not support encodings other than Latin1...

Reply:
```suggestion
        auto errorMsgTemplate = "[ WARN ]  - Failed to send text '%1' to the game. Giving up. Please change encoding to retry.";
```

Reply:
It does send to the game though, the message is not prohibited: but what the game will receive is rubbish. Some tend to ignore it outright so you won't notice. 

Changing the encoding isn't likely to help:

> While technically the user could change the encoding to something that works,
most of the time if the server supports something other than ASCII - say utf8 - it would have
preconfigured it using telnet CHARSET anyway, so telling the user to change encodings (something they
have no idea about) isn't helpful.

Reply:
I understand @SlySven's need for correctness, which is _excellent_ in code, but the computer-human interaction should not be treated as code: needs to be far more gentle, forgiving, and understandable.

Reply:
"[ WARN ]  - Can never send text '%1' correctly. The game won't understand it." 🤔

Reply:
That's a bit intense, I reckon.

Hence the more casual tone of the original message: it's an fyi.

Reply:
> How to flag then that there is a potentially on-going data loss - if we are shortening the message to not mention that it is the only time the warning will be given...

They'll know after the first message, if they can understand it. This is also a very much a theoretical problem: it is not a practical one we've seen users struggling with (first one was only reported recently after the code being there for a while).

Reply:
It is only a theoretical problem when it doesn't happen - when it does it has the potential to really mess up the end-user's experience. Finding a way to convey (and to continue to warn) that there is likely to be miscommunication -  a miss-match between what the user thinks they sending and what is *actually* going out over the wire is vital ***because it won't be obvious what exactly is happening***.

Reply:
> it has the potential to really mess up the end-user's experience

We will notice if people show up and start asking about this.

> it won't be obvious what exactly is happening

Nowadays people tend to ignore the 5 line message completely. 

Maybe this improves that after all?

Reply:
> They'll know after the first message, if they can understand it. ...

As it happens - I made that message a one shot precisely so that it does ***not*** spam the user each time it occurred - if they didn't understand and react to it then they won't get further messages - the data sent will continue to be corrupted and they will only get whatever messages the Game Server sends in response to those corrupted messages.

> ... Some [Game Servers] tend to ignore it outright so you won't notice. ...

In which case the messages will just *disappear* without trace!

If it makes things simpler then we *could* shorten the message - ideally to a one-liner - and show it *each* time a message tried to send unencode-able graphemes - that would make the code simpler in fact as then we do not have to flag whether a warning has been issued and to reset that flag when the encoding is changed...!